src/test/java/bio/overture/ego/service/RefreshContextServiceTest.java

```diff
@@ -36,11 +37,34 @@
   @Rule public ExpectedException exceptionRule = ExpectedException.none();
 
   @Test
-  public void refresh_usedTokenIsDeleted() {
+  public void refresh_invalidContext_usedTokenIsDeleted() {
+    val user1 = entityGenerator.setupUserWithRefreshToken("User One");
+    // force invalid refreshContext with invalid user status
+    user1.setStatus(StatusType.PENDING);
+    val user1Token = tokenService.generateUserToken(user1);
+    val refreshToken1 = user1.getRefreshToken();
+
+    Assert.assertTrue(refreshContextService.getById(refreshToken1.getId()) != null);
+    val incomingRefreshContext =
+        refreshContextService.createRefreshContext(refreshToken1.getId().toString(), user1Token);
+    refreshContextService.disassociateUserAndDelete(user1Token);
+
+    exceptionRule.expect(NotFoundException.class);
+    exceptionRule.expectMessage(
+        String.format("RefreshToken '%s' does not exist", refreshToken1.getId()));
+    Assert.assertTrue(refreshContextService.get(refreshToken1.getId(), false) == refreshToken1);
```

Comment:
this is a java gotcha! Read below
https://stackoverflow.com/a/10806192/1952450

basically, you have to use `.equals()` for this type of comparison

In a nut shell, the following code runs without failure/errors:
```java
RefreshToken r1 = RefreshToken.builder().id("someId").build();
RefreshToken r2 = r1;
RefreshToken r3 = RefreshToken.builder().id("someId").build();
RefreshToken r4 = RefreshToken.builder().id("SOMETHING_ELSE").build();

assertTrue(r1 == r2);
assertTrue(r1 != r3);
assertTrue(r1 != r4);
assertTrue(r2 != r3);
assertTrue(r2 != r4); 
assertTrue(r3 != r4);

assertTrue(r1.equals(r2));
assertTrue(r1.equals(r3));
assertFalse(r1.equals(r4));
assertTrue(r2.equals(r3));
assertFalse(r2.equals(r4));
assertFalse(r3.equals(r4));
```

For the selected line of code, you can use:
```java
assertEquals(refreshContextService.get(refreshToken1.getId(), false), refreshToken1);
```
which calls the .equals method under the hood.

Reply:
Aha...i ran into this with a string comparison but didn't realize it here!